src/types/errorType.ts

```diff
@@ -0,0 +1,26 @@
+export interface CardError {
+  cardNumbersError: CardNumbersError;
+  expirationDateError: ExpirationDateError;
+  ownerError: ErrorType;
+}
+
+export interface ErrorType {
+  hasError: boolean;
+  errorMessage: string;
+}
+
+export interface CardNumberBlockError extends ErrorType {
+  isDisable: boolean;
+}
+
+export interface ExpirationDateError {
+  month: ErrorType;
+  year: ErrorType;
+}
+
+export interface CardNumbersError {
+  firstBlock: CardNumberBlockError;
+  secondBlock: CardNumberBlockError;
+  thirdBlock: CardNumberBlockError;
+  fourthBlock: CardNumberBlockError;
+}
```

Comment:
여기도 인터페이스를 계층화해서 관리하면 조금 더 깔끔하게 사용할 수 있을 것 같아요!
또한, CardNumberBlockError는 ErrorType을 상속받는데 서로 이름의 형식이 달라 코드를 처음 보는 사람은 정확한 상속 관계를 유추하기 어려울 수 있을 것 같아요!

```suggestion
export interface ErrorType {
  hasError: boolean;
  errorMessage: string;
}

export interface CardNumberErrorType extends ErrorType {
  isDisable: boolean;
}

export interface CardFormError {
  cardNumbers: {
    firstBlock: CardNumberErrorType;
    secondBlock: CardNumberErrorType;
    thirdBlock: CardNumberErrorType;
    fourthBlock: CardNumberErrorType;
  };
  expiration: {
    month: ErrorType;
    year: ErrorType;
  };
  owner: ErrorType;
}
```